### DISCUSSION TITLE
How to change window size and position, mouse position with VSG tools?

### DISCUSSION THREAD

**Discussion by Anshi701** [vsg-dev/VulkanSceneGraph#1018] [Q&A]
Hello, i tried to make some user control to window size and position and mouse position in RUNTIME after initialization, but couldnt find some way to do that. Can you please tell me if there is posibility to do that, and if yes, how can it be achieved?

**Comment by WdsPjm** [comment]
![image](https://github.com/vsg-dev/VulkanSceneGraph/assets/52599568/1d40657e-b15f-4ca5-b116-fa284c1ae574)

**Reply by Anshi701** [reply]
Sorry, i need to correct my question - i was meaning in runtime, after initialization done.

**Reply by robertosfield** [reply]
I have followed a policy of implement features when they are required, to avoid feature bloat where the code base gets clogged up with features that no one uses.  5 1/2 years since the VulkanSceneGraph's inception and we haven't yet implemented changing the window size or mouse position at runtime :-)

There are 2 approaches come to mind:

1. use underlying windowing API calls to change the window size or mouse position
2. implement 1 but integrate this functionality in the the VSG's existing native windowing support and add a couple of methods in to vsg::Window to invoke this functionality.

The OpenSceneGraph project took approach 2, and I'd be happy for the VSG to follow suite, I personally only have Linux dev system so can only help out with implementation and testing on Xcb, others in the community would have to step up and implement the other platforms.  I'm really busy with other tasks right now so can't step forward an help out with Xcb implementation right away. 

Eric Raymond coined the phrase "scratch the itch"  and this has always worked well for the OpenSceneGraph and VulkanSceneGraph:   If you haven't read it already have a read of his [The_Cathedral_and_the_Bazaar ](https://en.wikipedia.org/wiki/The_Cathedral_and_the_Bazaar) paper.

**Reply by Anshi701** [reply]
Alright, i got it. Thank you for responce! Hope it will be inplemented in the near future.

**Reply by robertosfield** [reply]
Where platform and APIs are you using?

As for "implemented in the near future" your the one who needs the feature right now so please don't rule yourself out as one that might step out and help implement it :-)
